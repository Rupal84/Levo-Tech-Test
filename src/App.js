import { useState } from 'react';
import Container from 'react-bootstrap/Container';
import Row from 'react-bootstrap/Row';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';
import CardDeck from 'react-bootstrap/CardDeck';
import './App.scss';

function App() {
  const [cardsCount, setCardsCount] = useState(3);
  const getCard = ({ title, content, date }, index) => {
    return (
      <Col key={index} xs={12} sm={12} md={6} lg={4} xl={4} className="card-col">
      <Card >
        <Card.Body>
        <p>{date}</p>
          <Card.Title>{title}</Card.Title>
          <Card.Text>
            {content}
          </Card.Text>

        </Card.Body>
        <Card.Footer>
          <button type="button" className="card-btn btn btn-outline-primary">READ MORE</button>
        </Card.Footer>
      </Card>
      </Col>
    )
  }
  
  const cards = [
    {
      title: "Love of learning, art keys to a great year for Gwen",
      content: "Nunc nec scelerisque felis. Nam porta eros odio, sit amet pretium lectus dapibus a. Nam in venenatis dolor. Etiam ac fringilla purus. Ut consequat. ",
      date: '30 Sep, 2020'
    },
    {
      title: "Curious mind leads the way for carer Jill",
      content: "Sed gravida tempor justo, accumsan suscipit ex pulvinar et. Mauris mi lorem, tempor sit amet feugiat in, efficitur eu mauris. Nulla facilisi. ",
      date: '23 Sep, 2020'
    },
    {
      title: "Celebrating our volunteers",
      content: "Curabitur ornare nisl quis pharetra tincidunt. Cras congue leo metus, eu accumsan ex vulputate vel. Proin elementum varius sapien, sit amet eleifend nunc iaculis eu eu accumsan ex vulputate.",
      date: '12 Sep, 2020'
    },
    {
      title: "A great birthday present",
      content: "Nulla commodo tincidunt lectus varius semper. Aenean interdum justo sit amet fringilla ullamcorper. Nulla pharetra ipsum nulla, sed ornare orci tincidunt et. ",
      date: '30 Jan, 2020'
    },
    {
      title: "A journey to nowhere",
      content: "Curabitur gravida ligula vitae pretium pellentesque. Curabitur ullamcorper, nibh non tempor laoreet, purus orci posuere tortor.",
      date: '23 Dec, 2020'
    },
    {
      title: "Amazing restaurants",
      content: "Aliquam pharetra arcu a ligula aliquam viverra. Proin sollicitudin nulla eros, eu varius tortor tincidunt vel. Sed vitae metus dolor.",
      date: '30 Nov, 2020'
    }
  ]
  return (
    <Container>
      <div className="header-container">
        <h1 className="display-4 header">Related Articles</h1>
        <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam felis lectus, lobortis vel eros nec, vulputate sodales quam. Aenean ultricies viverra dui nec ullamcorper. Integer ultricies sodales ex quis gravida. Integer quis risus hendrerit, faucibus libero sit amet, egestas elit. Etiam porta lectus malesuada dolor tempor, et tristique risus pulvinar. </p>
      </div>
      <Row>
        <Col>
          <CardDeck>
            <Row>
            {cards.slice(0, cardsCount).map((card, index) => getCard(card, index))}
            </Row>
          </CardDeck></Col>
      </Row>
      <div className="readmore-btn">
  <button type="button" className="btn btn-outline-danger" 
  onClick={()=> {
    cardsCount === 3 ? setCardsCount(6) : setCardsCount(3);
   }}>{cardsCount === 3 ? 'READ MORE' : 'SHOW LESS'}</button>
      </div>
    </Container>
  );
}

export default App;
