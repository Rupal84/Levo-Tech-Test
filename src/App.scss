.header {
  font-weight: 400;
  margin: 16px 0;
  color: #d11010;
}

.header-container {
  max-width: 600px;
  margin-bottom: 40px;
}

.card-footer {
  background: none;
  border: none;
}

.card-footer > button.btn-primary{
  background: none;
}
.readmore-btn {
  padding: 30px 0;
  display: flex;
  justify-content: center;
  border-color: white;
}
.card-deck .row > div {
  margin-bottom: 10px;
}

.btn-outline-primary {
  border-color: white;
  color: white;
}

.card-deck {
  .row {
    div {
      .card {
        .card-title {
          width: 100%
        }
        background: #ffc107;
        color: white;
        min-height: 344px;
    }
    }
    @media screen and (min-width: 412px) and  (max-width:768px){
      div:nth-of-type(2n+1) {
        div {
          background: #ffc107;
        }
      }
      div:nth-of-type(2n) {
        div {
          background: #9189bd;
        }
      }      
    }
    @media screen and (min-width:769px){
      div:nth-of-type(3n+1) {
        div {
          background: #ffc107;
        }
      }
      div:nth-of-type(3n+2) {
        div {
          background: #9189bd;
        }
      }
      div:nth-of-type(3n) {
        div {
          background: #8a6891;
        }
      }
    }
  }
}
.card-col {
  padding: 10px 0;
}

.container > .row {
  width: 100%;
  margin: auto;
}


