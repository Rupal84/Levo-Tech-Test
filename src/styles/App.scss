@import "./variables.scss";

.header {
  font-weight: 400;
  margin: 2rem 0;
  color: map-get($colors, bright-red);
}

.header-container {
  max-width: 40rem;
  margin-bottom: 2rem;
}

.card-footer {
  background: none;
  border: none;
}

.card-footer > button.btn-primary {
  background: none;
}

.readmore-btn {
  padding: 1rem 0;
  display: flex;
  justify-content: center;
  border-color: map-get($colors, custom-wht);

  .btn {
    font-weight: 500;
  }
}

.btn-outline-primary {
  border-color: map-get($colors, custom-wht);
  color: map-get($colors, custom-wht);
  font-weight: 500;
}

.card-deck {
  .row {
    div {
      .card {
        .card-title {
          width: 100%;
        }

        margin-bottom: 0.7rem;
        background: map-get($colors, bright-yellow);
        color: map-get($colors, custom-wht);
        min-height: 344px;
      }
    }

    @media screen and (min-width: map-get($breakpoints, tablet-min)) and  (max-width: map-get($breakpoints, tablet-max)) {
      div:nth-of-type(2n+1) div {
        background: map-get($colors, bright-yellow);
      }

      div:nth-of-type(2n) div {
        background: map-get($colors, light-blue);
      }
    }

    @media screen and (min-width: map-get($breakpoints, desktop)) {
      div:nth-of-type(3n+1) div {
        background: map-get($colors, bright-yellow);
      }

      div:nth-of-type(3n+2) div {
        background: map-get($colors, light-blue);
      }

      div:nth-of-type(3n) div {
        background: map-get($colors, dark-purple);
      }
    }
  }
}

.card-col {
  padding: 0.5rem 0;
}

.container > .row {
  width: 100%;
  margin: auto;
}
