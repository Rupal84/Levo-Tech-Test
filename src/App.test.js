import React from 'react';
import { fireEvent, render } from '@testing-library/react';
import App from './App';

describe('App Component', () => {
  test('renders component successfully', () => {
    const {container} = render(<App />);  
    expect(container).toMatchSnapshot();
  });
  
  test('Click Read more button and display more cards', () => {
    const {getByTestId, getAllByTestId} = render(<App />);  
    const btn = getByTestId('read-more');
    fireEvent.click(btn);
    expect(getAllByTestId('card')).toHaveLength(6);
  });
})

