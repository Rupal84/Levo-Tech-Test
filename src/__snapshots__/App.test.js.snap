// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders component successfully 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="header-container"
    >
      <h1
        class="display-4 header"
      >
        Related Articles
      </h1>
      <p>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aliquam felis lectus, lobortis vel eros nec, vulputate sodales quam. Aenean ultricies viverra dui nec ullamcorper. Integer ultricies sodales ex quis gravida. Integer quis risus hendrerit, faucibus libero sit amet, egestas elit. Etiam porta lectus malesuada dolor tempor, et tristique risus pulvinar. 
      </p>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="card-deck"
        >
          <div
            class="row"
          >
            <div
              class="card-col col-xl-4 col-lg-4 col-md-6 col-sm-12 col-12"
            >
              <div
                class="card"
                data-testid="card"
              >
                <div
                  class="card-body"
                >
                  <p>
                    30 Sep, 2020
                  </p>
                  <div
                    class="card-title h5"
                  >
                    Love of learning, art keys to a great year for Gwen
                  </div>
                  <p
                    class="card-text"
                  >
                    Nunc nec scelerisque felis. Nam porta eros odio, sit amet pretium lectus dapibus a. Nam in venenatis dolor. Etiam ac fringilla purus. Ut consequat. 
                  </p>
                </div>
                <div
                  class="card-footer"
                >
                  <button
                    class="card-btn btn btn-outline-primary"
                    type="button"
                  >
                    READ MORE
                  </button>
                </div>
              </div>
            </div>
            <div
              class="card-col col-xl-4 col-lg-4 col-md-6 col-sm-12 col-12"
            >
              <div
                class="card"
                data-testid="card"
              >
                <div
                  class="card-body"
                >
                  <p>
                    23 Sep, 2020
                  </p>
                  <div
                    class="card-title h5"
                  >
                    Curious mind leads the way for carer Jill
                  </div>
                  <p
                    class="card-text"
                  >
                    Sed gravida tempor justo, accumsan suscipit ex pulvinar et. Mauris mi lorem, tempor sit amet feugiat in, efficitur eu mauris. Nulla facilisi. 
                  </p>
                </div>
                <div
                  class="card-footer"
                >
                  <button
                    class="card-btn btn btn-outline-primary"
                    type="button"
                  >
                    READ MORE
                  </button>
                </div>
              </div>
            </div>
            <div
              class="card-col col-xl-4 col-lg-4 col-md-6 col-sm-12 col-12"
            >
              <div
                class="card"
                data-testid="card"
              >
                <div
                  class="card-body"
                >
                  <p>
                    12 Sep, 2020
                  </p>
                  <div
                    class="card-title h5"
                  >
                    Celebrating our volunteers
                  </div>
                  <p
                    class="card-text"
                  >
                    Curabitur ornare nisl quis pharetra tincidunt. Cras congue leo metus, eu accumsan ex vulputate vel. Proin elementum varius sapien, sit amet eleifend nunc iaculis eu eu accumsan ex vulputate.
                  </p>
                </div>
                <div
                  class="card-footer"
                >
                  <button
                    class="card-btn btn btn-outline-primary"
                    type="button"
                  >
                    READ MORE
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="readmore-btn"
    >
      <button
        class="btn btn-outline-danger"
        data-testid="read-more"
        type="button"
      >
        READ MORE
      </button>
    </div>
  </div>
</div>
`;
